squad/frontend/failures.py

```diff
@@ -0,0 +1,48 @@
+from django.shortcuts import render
+
+from squad.http import auth
+from squad.frontend.views import get_build
+
+
+@auth
+def failures(request, group_slug, project_slug, build_version):
+    project = request.project
+    build = get_build(project, build_version)
+    environments = project.environments.order_by("slug")
+
+    search = request.GET.get('search', '')
+
+    if search:
+        tests = build.tests.filter(
+            result=False,
+            has_known_issues=False,
+            metadata__name__icontains=search,
+        ).prefetch_related('environment')
+    else:
+        tests = build.tests.filter(
+            result=False,
+            has_known_issues=False,
+        ).prefetch_related('environment')
```

Comment:
you can avoid this duplication by chaining filters, e.g.

```python
tests = build.tests.filter(
            result=False,
            has_known_issues=False,
).prefetch_related('environment')
if search:
    tests = tests.filter(metadata__name_icontains=search)
```

Reply:
Thanks. Fixed.